File: src/Other/Converters/RgbToBrushConverter.fs
namespace OpenSilver.Samples.Showcase

open System
open System.Globalization
open System.Windows.Data
open System.Windows.Media

/// Converts three RGB values to a SolidColorBrush.
type RgbToBrushConverter() =
    interface IMultiValueConverter with
        member _.Convert(values: obj[], targetType: Type, parameter: obj, culture: CultureInfo) : obj =
            match values with
            | [| :? double as r; :? double as g; :? double as b |] ->
                let color = Color.FromRgb(byte r, byte g, byte b)
                SolidColorBrush(color) :> obj
            | _ ->
                SolidColorBrush(Colors.Black) :> obj

        member _.ConvertBack(value: obj, targetTypes: Type[], parameter: obj, culture: CultureInfo) : obj[] =
            raise (NotImplementedException())

File: src/Other/Converters/RgbToBrushConverter.vb
Imports System.Globalization
Imports System.Windows.Data
Imports System.Windows.Media

Namespace OpenSilver.Samples.Showcase

    ''' <summary>
    ''' Converts three RGB values to a SolidColorBrush.
    ''' </summary>
    Public Class RgbToBrushConverter
        Implements IMultiValueConverter

        Public Function Convert(values As Object(), targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IMultiValueConverter.Convert
            If values.Length = 3 AndAlso
               TypeOf values(0) Is Double AndAlso
               TypeOf values(1) Is Double AndAlso
               TypeOf values(2) Is Double Then

                Dim r As Byte = CByte(values(0))
                Dim g As Byte = CByte(values(1))
                Dim b As Byte = CByte(values(2))

                Return New SolidColorBrush(Color.FromRgb(r, g, b))
            End If

            Return New SolidColorBrush(Colors.Black)
        End Function

        Public Function ConvertBack(value As Object, targetTypes As Type(), parameter As Object, culture As CultureInfo) As Object() Implements IMultiValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function

    End Class

End Namespace
